lib/backup/gcsremote/gcs.go

```diff
@@ -281,6 +290,9 @@ func (fs *FS) CreateFile(filePath string, data []byte) error {
 		_ = w.Close()
 		return fmt.Errorf("wrong data size uploaded to %q at %s (remote path %q); got %d bytes; want %d bytes", filePath, fs, o.ObjectName(), n, len(data))
 	}
+	if err := fs.maybeSetMetadata(o); err != nil {
```

Comment:
Looks like, there is more efficient way how to set metadata - instead of calling `Update` on objects. It's possible to modify attributes on `NewWriter` and `CopierFrom` objects:

```
	w := o.NewWriter(fs.ctx)
         w.ObjectAttrs = storage.ObjectAttrs{
    Metadata: map[string]string{},
  }
```

```
	copier := dstObj.CopierFrom(srcObj)
  copier.ObjectAttrs = storage.ObjectAttrs{
    Metadata: map[string]string{},
  }
```